shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/web/dto/mapper/ExaminationToStudyDTOMapper.java

```diff
@@ -0,0 +1,112 @@
+import org.shanoir.ng.shared.repository.SubjectRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This class maps the Examination objects from the Shanoir-NG database
+ * to a DICOM-study DTO to implement the DICOMweb protocol.
+ * 
+ * @author mkain
+ *
+ */
+@Mapper(componentModel = "spring")
+public abstract class ExaminationToStudyDTOMapper {
+	
+	@Autowired
+	protected SubjectRepository subjectRepository;
+	
+	public StudyDTO examinationToStudyDTO(Examination examination) {
+		StudyDTO studyDTO = new StudyDTO();
+		final String studyInstanceUID = UIDGeneration.ROOT + ".1." + examination.getId();
+		// 5 DICOM-study specific values
+		studyDTO.setStudyInstanceUID(studyInstanceUID);
+		studyDTO.setStudyID(examination.getId());
+		studyDTO.setStudyDescription(examination.getComment());
+//		studyDTO.setStudyDate(examination.getExaminationDate().toString());
+		studyDTO.setStudyDate("20220303");
+		studyDTO.setStudyTime("000000"); // today we do not store this info in our db
+		studyDTO.setAccessionNumber("");
+		// 4 patient specific values
+		// @TODO optimize here: not ask the database for each subject id, use a cached list?
+		Optional<Subject> subjectOpt = subjectRepository.findById(examination.getSubjectId());
+		String subjectName = "error_subject_name_not_found_in_db";
+		if (subjectOpt.isPresent()) {
+			subjectName = subjectOpt.get().getName();
+		}
+		studyDTO.setPatientName(subjectName);
+		studyDTO.setPatientID(examination.getSubjectId().toString());
+		studyDTO.setPatientBirthDate("01011960"); // @TODO not yet in ms datasets database
+		studyDTO.setPatientSex("F"); // @TODO not yet in ms datasets database
+		
+		addSeries(examination, studyDTO);
+		studyDTO.setNumInstances(1);
+		studyDTO.setModalities(studyDTO.getSeries().get(0).getModality());
+		
+		return studyDTO;
+	}
+
+	/**
+	 * This method transforms dataset acquisitions in Shanoir back
+	 * into DICOM series for the purpose of supporting the DICOMWeb protocol.
+	 * 
+	 * @param examination
+	 * @param studyDTO
+	 */
+	private void addSeries(Examination examination, StudyDTO studyDTO) {
+		List<SerieDTO> series = new ArrayList<SerieDTO>();
+		List<DatasetAcquisition> acquisitions = examination.getDatasetAcquisitions();
+		for (DatasetAcquisition datasetAcquisition : acquisitions) {
+			SerieDTO serie = new SerieDTO();
+			final String serieInstanceUID = UIDGeneration.ROOT + ".2." + datasetAcquisition.getId();
+			serie.setSerieInstanceUID(serieInstanceUID);
+			serie.setSeriesNumber(datasetAcquisition.getSortingIndex());
+			serie.setModality(datasetAcquisition.getType().toUpperCase());
+			
+			List<InstanceDTO> instances = new ArrayList<InstanceDTO>();
+			List<Dataset> datasets = datasetAcquisition.getDatasets();
+			for (Dataset dataset : datasets) {
+				InstanceDTO instance = new InstanceDTO();
+				MetadataDTO metadata = new MetadataDTO();
+				metadata.setSopClassUID("1.2.840.10008.5.1.4.1.1.2");
```

Comment:
Is it normal that this value is fixed ?

Reply:
no, but please as stated in my JavaDoc please ignore DICOMJson